subsystems/drivebase/SwerveDrive.java

```diff
@@ -311,8 +329,43 @@ public void mustangPeriodic() {
         Logger.recordOutput(DRIVEBASE_PITCH, getPitch());
         Logger.recordOutput(DRIVEBASE_ROLL, getRoll());
 
+        if (kConfig.kDriveMotorType.equals(Motor_Type.KRAKEN_X60)) {
+            double latestDriveCurrentDraw = 0;
+            for (int i = 0; i < mModules.length; i++) {
+                latestDriveCurrentDraw += ((TalonFX) mModules[i].getDriveMotor()).getSupplyCurrent();
+            }
+
+            double currentTime = System.currentTimeMillis() / 1000;  // Current time in seconds
+            driveCurrentBuffer.add(new CurrentMeasurement(currentTime, latestDriveCurrentDraw));
+            
+            // Remove old measurements from the buffer.
+            while (!driveCurrentBuffer.isEmpty() && currentTime - driveCurrentBuffer.peek().getTime() > 5) {
```

Comment:
we can simplify this. we don't care if it's exactly 5 seconds. this code will get called approx. 50 times per second so let's err on the side of efficiency. It's likely fine to just remove the oldest entry once we've hit enough samples and assume we're still at about a 5-second window - doing it over 250 samples is likely good enough (by the way, 5 should be a constant somewhere that we can change in case we decide 5 isn't the right size for the window). And if we just remove a single entry, then it's easy to calculate the new average without having to sum the whole window, right?

Reply:
For more efficiency: if we just keep a window that's big enough to store a certain number of samples (aka an array) and we don't care about the timestamp of a given sample, we could just store doubles and stop creating garbage by not instantiating more objects. If we don't create more garbage, we reduce the changes of the GC interrupting the program at an unfortunate time. Just keep track of where we currently are in the array, right?